### ISSUE TITLE
Comments Inside A Record Definition

### ISSUE THREAD

**Issue by s0kil** [mint-lang/mint#338]
This code throws a compiler error:
```mint
record LinkedList {
  head : Int,
  /* tail : LinkedList */
}
```
```
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

░ SYNTAX ERROR ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░

The body of a record definition must end with a closing bracket. 

I was looking for the bracket "}" but found "/*" instead. 

Here is the relevant code snippet: 

┌─┬ source/Main.mint:3:3 ──────────────────────────────────────────────────────┐
│1│ record LinkedList {                                                        │
│2│   head : Int,                                                              │
│3│   /* tail : LinkedList */                                                  │
│4│ }                                                                          │
│5│                                                                            │
│6│ component Main {                                                           │
│7│   style base {                                                             │
└─┴────────────────────────────────────────────────────────────────────────────┘

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

**Comment by Namek** [comment]
isn't it because of the comma?

**Comment by gdotdesign** [comment]
> isn't it because of the comma?

No it's because it's not implemented :joy: 

The parser doesn't have a tokenization phase so comments are added everywhere manually. 

So basically this line: https://github.com/mint-lang/mint/blob/master/src/parsers/record_definition.cr#L23
should look like this: `) { record_definition_field || comment }.compact`. 

Then the type of the fields will be `Array(RecordDefnitionField | Comment)` and because of that the node type, type checker, compiler and formatter needs to be updated accordingly.

**Comment by Namek** [comment]
@gdotdesign is the performance an issue of not having the tokenization phase?

**Comment by gdotdesign** [comment]
> @gdotdesign is the performance an issue of not having the tokenization phase?

I don't understand the question, please clarify :slightly_smiling_face:

**Comment by Namek** [comment]
@gdotdesign rephrasing: why is there no tokenization phase?

**Comment by gdotdesign** [comment]
> @gdotdesign rephrasing: why is there no tokenization phase?

It's two fold:
1. I didn't study compilers before and wasn't aware of the tokanization phase when I created Mint
2. I wanted precise control over the error messages (I used parse generators before and they have limited ways to manage errors)

**Comment by gdotdesign** [comment]
Fixed in e5becf4d8c2eb84b2ad81331160890eb8e2663fd
